File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode list = new ListNode();
        ListNode dummy = list;

        ListNode head1 = list1;
        ListNode head2 = list2;

        while (head1 != null && head2 != null){

            if (head1.val < head2.val){
                dummy.next = head1;
                head1 = head1.next;
            }else{
                dummy.next = head2;
                head2 = head2.next;
            }

            dummy = dummy.next;

        }

        if (head1 != null){
            dummy.next = head1;
        }
        else if (head2 != null){
            dummy.next = head2;
        }

        return list.next;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        
        l = 0 

        curr = head

        while curr:

            l +=1
            curr = curr.next

        distance =  l - n

        if distance == 0:
            return head.next

        curr = head

        i = 0

        while i < distance:

            i +=1

            if i == distance:
                curr.next = curr.next.next
                break
            else:
                curr = curr.next
        return head